File: modules/cubex-config/src/main/java/org/cubexmc/config/MigrationContext.java
package org.cubexmc.config;

import java.io.File;
import org.bukkit.configuration.file.YamlConfiguration;

public interface MigrationContext {
    File file();

    String resourcePath();

    YamlConfiguration yaml();

    void warning(String path, String message);

    void fail(String path, String message);
}

File: modules/cubex-config/src/main/kotlin/org/cubexmc/config/MigrationContext.kt
package org.cubexmc.config

import org.bukkit.configuration.file.YamlConfiguration
import java.io.File

interface MigrationContext {
    fun file(): File
    fun resourcePath(): String
    fun yaml(): YamlConfiguration
    fun warning(path: String?, message: String?)
    fun fail(path: String?, message: String?)
}
